src/encoded/commands/release_file.py

```diff
@@ -0,0 +1,491 @@
+    SUBMITTER_ID = "submitter_id"
+    SUBMITTED_ID = "submitted_id"
+    TYPE = "@type"
+    UPLOADED = "uploaded"
+    UUID = "uuid"
+    VARIANT_TYPE = "variant_type"
+    VERSION = "version"
+
+
+# dataset is required but comes in through input args for now
+REQUIRED_FILE_PROPS = [PC.SEQUENCING_CENTER]
+
+
+class FileRelease:
+
+    def __init__(self, auth_key):
+        self.key = auth_key
+        self.patch_infos = []
+        self.patch_dicts = []
+        self.warnings = []
+        self.file_accession = ""
+
+    def prepare(self, file_identifier: str, dataset: str):
+
+        file = self.get_metadata(file_identifier)
+        self.file_accession = file[PC.ACCESSION]
+        self.check_file_validity(file)
+        fileset = self.get_fileset_from_file(file)
+        self.add_file_patchdict(file, fileset, dataset)
+
+        quality_metrics = self.get_quality_metrics_from_file(file)
+        self.add_release_items_to_patchdict(quality_metrics, "QualityMetric")
+
+        quality_metrics_zips = self.get_quality_metrics_zip_files(quality_metrics)
+        self.add_release_items_to_patchdict(
+            quality_metrics_zips, f"Compressed QC metrics file"
+        )
+
+        # Get higher level items starting from file set in order to set them to released
+        self.add_release_item_to_patchdict(
+            fileset, f"FileSet - {fileset[PC.SUBMITTED_ID]}"
+        )
+        sequencing = self.get_metadata(fileset[PC.SEQUENCING])
+        self.add_release_item_to_patchdict(
+            sequencing, f"Sequencing - {sequencing[PC.SUBMITTED_ID]}"
+        )
+
+        if len(fileset[PC.LIBRARIES]) > 1:
+            self.add_warning(
+                f"Multiple libraries attached to file set {fileset[PC.ACCESSION]}"
+            )
+
+        for library_id in fileset[PC.LIBRARIES]:
+            library = self.get_metadata(library_id)
+            self.add_release_item_to_patchdict(
+                library, f"Library - {library[PC.SUBMITTED_ID]}"
+            )
+
+            assay = self.get_metadata(library[PC.ASSAY])
+            self.add_release_item_to_patchdict(assay, f"Assay - {assay[PC.IDENTIFIER]}")
+
+            analyte = self.get_metadata(library[PC.ANALYTE])
+            self.add_release_item_to_patchdict(
+                analyte, f"Analyte - {analyte[PC.SUBMITTED_ID]}"
+            )
+
+            for sample_uuid in analyte[PC.SAMPLES]:
+                sample = self.get_metadata(sample_uuid)
+                self.add_release_item_to_patchdict(
+                    sample, f"Sample - {sample[PC.SUBMITTED_ID]}"
+                )
+
+                sample_sources = sample[PC.SAMPLE_SOURCES]
+                for sample_source_id in sample_sources:
+                    sample_source = self.get_metadata(sample_source_id)
+                    self.add_release_item_to_patchdict(
+                        sample_source,
+                        f"SampleSource - {sample_source[PC.SUBMITTED_ID]}",
+                    )
```

Comment:
Probably need to go more layers up here, as sample sources can be a tissue or a cell culture (mixture). For tissue, would want to ensure the donor is released, for cell culture the cell line, and for a cell culture mixture its cell cultures and corresponding cell lines.

Reply:
Done.